test/sets.jl

```diff
@@ -452,6 +452,7 @@ end
     @test allunique(7:-1:1)       # negative step
     @test allunique(Date(2018, 8, 7):Day(1):Date(2018, 8, 11))  # JuliaCon 2018
     @test allunique(DateTime(2018, 8, 7):Hour(1):DateTime(2018, 8, 11))
+    @test allunique(('a':1:'c')[1:2]) == true
```

Comment:
I would use `'a':'\x01':'b'` instead of (or in addition to) `('a':1:'c')[1:2]`, to be sure what the test is actually testing, and doesn't depend on possible future changes of the result type of `('a':1:'c')[1:2]`.

Reply:
Wouldn't that still error even after this change? Since the step type is `Char` which correctly does not have a zero element and also is not a valid step type. Perhaps I'm misunderstanding the suggestion.

Reply:
Oh I got it wrong. I don't fully understand what happens here, but I just did:
```
julia> ('a':1:'c')[1:2]
'a':'\x01':'b'
```
And assumed `allunique('a':'\x01':'b')` was the problem, but `'a':'\x01':'b'` is actually not valid input:
```julia
julia> 'a':'\x01':'b'
ERROR: MethodError: no method matching zero(::Char)
```
Forget my comment above and carry on, though there might be a problem with the printing of `('a':1:'c')[1:2]`.

Reply:
Ok, I found the source of the confusion: until very recently, `step(::StepRangeLen)` was returning a `Char`, hence the output above `'a':'\x01':'b'`, now `step` properly doesn't convert the step to `Char` (cf. #34412), and keeps it as an `Int`, so we now have:
```julia
julia>  ('a':1:'c')[1:2]
'a':1:'b'
```
So at least now `('a':1:'c')[1:2] == 'a':1:'b'` even though the types don't match.